src/main/java/fr/polytech/sim/transport/Vehicle.java

```diff
@@ -2,17 +2,16 @@
 
 import fr.polytech.sim.log.ConsoleLogger;
 import fr.polytech.sim.log.Logger;
-import java.util.HashSet;
-import java.util.Iterator;
-import java.util.Set;
+
+import java.util.*;
 
 /**
  * Vehicle abstraction.
  */
 public abstract class Vehicle implements MobileObject {
 
     private final Logger logger = new ConsoleLogger("Vehicle");
-    protected final Set<MobileObject> components = new HashSet<>();
+    protected final ArrayList<MobileObject> components = new ArrayList<MobileObject>();
```

Comment:
```suggestion
    protected final List<MobileObject> components = new ArrayList<>();
```
* C'est toujours préférable d'avoir un type de déclaration le plus géneral possible.
* Pas besoin de confirmer le paramètre de type à la création (diamond operator).